src/commands/moderation/move.js

```diff
@@ -0,0 +1,157 @@
+    const embed = new SwitchbladeEmbed(author)
+    if (!matchedLink) {
+      embed.setTitle(t('commands:move.invalidMessageLink'))
+      embed.setDescription(t('commands:move.missingMessageLink'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    const guildId = matchedLink[1]
+    const channelId = matchedLink[2]
+    const messageId = matchedLink[3]
+
+    if (guildId !== guild.id) {
+      embed.setDescription(t('commands:move.onlyThisGuild'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    if (!guild.channels.has(channelId)) {
+      embed.setDescription(t('commands:move.channelDoesntExist'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    const channelFrom = guild.channels.get(channelId)
+
+    if (!channelFrom) {
+      embed.setDescription(t('commands:move.couldntFindChannel'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    if (!destinationChannel.permissionsFor(this.client.user.id).has('SEND_MESSAGES')) {
+      embed.setDescription(t('commands:move.cantSendInChannel'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    try {
+      const recievedMessage = await channelFrom.fetchMessage(messageId)
+
+      if (!recievedMessage) {
+        embed.setDescription(t('commands:move.couldntFindMessage'))
+        embed.setColor(Constants.ERROR_COLOR)
+        channel.send(embed)
+        return
+      }
+    } catch (e) {
+      embed.setDescription(t('commands:move.couldntFindMessage'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    /*
+    Doing this outside the try catch made the handler on line 71 get ignored.
+    While doing this looks weird, but it works. ~ Zerinho6.
+    */
+    const recievedMessage = await channelFrom.fetchMessage(messageId)
+
+    if (recievedMessage.channel.nsfw && !destinationChannel.nsfw) {
+      embed.setDescription(t('commands:move.channelsHaveDifferentType'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    if (!recievedMessage.channel.permissionsFor(this.client.user.id).has('MANAGE_MESSAGES')) {
+      embed.setDescription(t('commands:move.cantDeleteMessage'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    if (!recievedMessage.channel.permissionsFor(author.id).has('SEND_MESSAGES')) {
+      embed.setDescription(t('commands:move.userCantSend'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    try {
+      destinationChannel.send(`${t('commands:move.messageSentBy')} ${recievedMessage.author.username} ${t('commands:move.movedFrom')} ${recievedMessage.channel.name}  ${t('commands:move.by')} ${author.username}`)
```

Comment:
what the fuck this can leave so many problems in translations. you need to do something like that on translations:
```
"thisMessage": "Sent by {{author}} moved from {{channel}} by {{something}}"
```
and then
```js
destinationChannel.send(t('commands:move.thisMessage', {
	author: recievedMessage.author.username,
	...
}))
```